src/jwk/index.ts

```diff
@@ -22,3 +24,32 @@ import type { JWK } from '../types';
  */
 export const thumbprint = (jwk: JWK): Promise<string> =>
   IoReactNativeJwt.thumbprint(jwk);
+
+/**
+ * Returns a JSON Web Key Set (JWKS) downloaded from a remote endpoint,
+ * that is, for example, an OAuth 2.0 or OIDC jwks_uri.
+ *
+ * @param url URL string to fetch the JSON Web Key Set from.
+ * @param appFetch fetch function (optional)
+ */
+export const getRemoteJWKSet = async (
+  url: string,
+  appFetch: GlobalFetch['fetch'] = fetch
+): Promise<JWK[]> => {
+  let response = await appFetch(url);
+
+  if (response.status === 200) {
+    let jwt = await response.text();
+    let { payload } = await decode(jwt);
+    let metadata = payload as unknown as GenericMetadata;
+    if (metadata.jwks && metadata.jwks.keys) {
+      return metadata.jwks.keys;
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue(polluted type-system):** This manual casting may lead to inconsistent data if `keys` are not in the `JWK` shape. Users of `getRemoteJWKSet‏` will assume the existence of fields that may not exist, leading to some unexpected production error. I think we must do either:
* parse the payload to the specific `GenericMetadata` shape, or
* return `Promise<unknown>` and let the user parse the result (and thus handle errors safely)

Reply:
I added integration with ZOD!